### ISSUE TITLE
#203 Setup basic CI pipeline - [merged]

### ISSUE THREAD

**Issue by noheton** [noheton/shepard#973]
***In GitLab by @xit-mohamed on Sep 5, 2024, 15:49 GMT+2:***

_Merges 203-Basic-ci-for-nuxtend -> develop_

# Merge Request

### :notebook: Checklist

- [x] The code compiles without any warnings.
- [x] I followed the [code review checklist](https://gitlab.com/dlr-shepard/shepard/-/blob/main/CONTRIBUTING.md#code-review-checklist).
- [x] The documentation has been added/updated.

### :link: Related Issues

- Related #<issue number>
- Closes #<issue number>

Closes #203

**Approved by:** xit-paul-reinhardt, xit-max-hey

*Migrated from GitLab: https://gitlab.com/dlr-shepard/shepard/-/merge_requests/254*

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 5, 2024, 15:53 GMT+2:***

added 1 commit

<ul><li>d7af23a8 - #203 Correct step name</li></ul>

[Compare with previous version](/dlr-shepard/shepard/-/merge_requests/254/diffs?diff_id=1111789350&start_sha=008855a5b97397f554ab27d533c4e8929c70e3e1)

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 5, 2024, 15:54 GMT+2:***

added 1 commit

<ul><li>02cda29b - #203 Fix artifact path name</li></ul>

[Compare with previous version](/dlr-shepard/shepard/-/merge_requests/254/diffs?diff_id=1111791494&start_sha=d7af23a8e374fbbbaae10b397571c00d1aafb099)

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 5, 2024, 16:12 GMT+2:***

added 1 commit

<ul><li>25966219 - #203 Switch to node 20</li></ul>

[Compare with previous version](/dlr-shepard/shepard/-/merge_requests/254/diffs?diff_id=1111820639&start_sha=02cda29b297ac1bc92011e90cf64f14c867523a8)

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 5, 2024, 16:14 GMT+2:***

added 1 commit

<ul><li>cbfd724e - #203 adapt the CONTRIBUTING.md</li></ul>

[Compare with previous version](/dlr-shepard/shepard/-/merge_requests/254/diffs?diff_id=1111823400&start_sha=259662197150b9bb4c600c8756b3e519ca5ac36e)

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 5, 2024, 16:22 GMT+2:***

added 1 commit

<ul><li>bf8ea241 - #203 restore old package-lock</li></ul>

[Compare with previous version](/dlr-shepard/shepard/-/merge_requests/254/diffs?diff_id=1111836114&start_sha=cbfd724eb85e03470274514f05310ba3d94fee02)

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 5, 2024, 16:28 GMT+2:***

added 1 commit

<ul><li>2046ccb5 - #203 Remove nuxtend/package-lock</li></ul>

[Compare with previous version](/dlr-shepard/shepard/-/merge_requests/254/diffs?diff_id=1111847339&start_sha=bf8ea2412f11d9e36a18b13500ccb073ac6a4b0d)

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 5, 2024, 16:36 GMT+2:***

added 1 commit

<ul><li>cad8612b - #203 Specify the correct prepare script</li></ul>

[Compare with previous version](/dlr-shepard/shepard/-/merge_requests/254/diffs?diff_id=1111861145&start_sha=2046ccb5a9b3d2fcb04f47a9baa143a038ff9cdb)

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 5, 2024, 16:56 GMT+2:***

marked this merge request as **ready**

**Comment by noheton** [comment]
**Discussion in GitLab:**

***In GitLab by @xit-paul-reinhardt on Sep 5, 2024, 16:58 GMT+2:***

Commented on [nuxtend/package.json line 21](https://github.com/noheton/shepard/compare/2cafb783d5f7984b3c6c2cbcf5bda03740c92d00...cad8612b70fcdd211b388f4d3f4a5586c1b837c4#diff-033da1de3bfc6479af1dedc7d275f1cd430b9c4da9055ed89750c60e8e8dae84R21)

https://github.com/noheton/shepard/blob/cad8612b70fcdd211b388f4d3f4a5586c1b837c4/nuxtend/package.json#L21



this version should be pinned

***In GitLab by @xit-mohamed on Sep 5, 2024, 17:08 GMT+2:***

changed this line in [version 9 of the diff](/dlr-shepard/shepard/-/merge_requests/254/diffs?diff_id=1111912107&start_sha=cad8612b70fcdd211b388f4d3f4a5586c1b837c4#5cece088e2b6d84390d050ec4b07d9db6ac67627_21_19)

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 5, 2024, 17:08 GMT+2:***

added 1 commit

<ul><li>18418242 - #203 remove unecessary scripts</li></ul>

[Compare with previous version](/dlr-shepard/shepard/-/merge_requests/254/diffs?diff_id=1111912107&start_sha=cad8612b70fcdd211b388f4d3f4a5586c1b837c4)

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Sep 5, 2024, 17:11 GMT+2:***

marked the checklist item **The code compiles without any warnings.** as completed

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Sep 5, 2024, 17:11 GMT+2:***

marked the checklist item **I followed the [code review checklist](https://gitlab.com/dlr-shepard/shepard/-/blob/main/CONTRIBUTING.md#code-review-checklist).** as completed

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Sep 5, 2024, 17:11 GMT+2:***

marked the checklist item **The documentation has been added/updated.** as completed

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Sep 5, 2024, 17:11 GMT+2:***

approved this merge request

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Sep 5, 2024, 17:12 GMT+2:***

approved this merge request

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 5, 2024, 17:13 GMT+2:***

added 1 commit

<ul><li>1422c577 - #203 Remove nuxtend/.prettierignore</li></ul>

[Compare with previous version](/dlr-shepard/shepard/-/merge_requests/254/diffs?diff_id=1111919600&start_sha=184182427d4dc4ae58cb94961eeee0bb03ba0591)

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 5, 2024, 17:13 GMT+2:***

reset approvals from @xit-paul-reinhardt and @xit-max-hey by pushing to the branch

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Sep 5, 2024, 17:13 GMT+2:***

approved this merge request

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 5, 2024, 17:14 GMT+2:***

enabled automatic add to merge train when checks pass

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Sep 5, 2024, 17:18 GMT+2:***

resolved all threads

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Sep 5, 2024, 17:18 GMT+2:***

approved this merge request

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 5, 2024, 17:20 GMT+2:***

started a [merge train](https://gitlab.com/dlr-shepard/shepard/-/merge_trains)

**Comment by noheton** [comment]
**Discussion in GitLab:**

***In GitLab by @thaase on Sep 6, 2024, 08:25 GMT+2:***

Commented on [CONTRIBUTING.md line 90](https://github.com/noheton/shepard/compare/2cafb783d5f7984b3c6c2cbcf5bda03740c92d00...1422c5770ff3c29b5fd205187a9f13b10130b81c#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055L90)

https://github.com/noheton/shepard/blob/1422c5770ff3c29b5fd205187a9f13b10130b81c/CONTRIBUTING.md#L90



This should be pinned in `renovate.json` and documented. However, I personally see no reason why we need to downgrade the client images. The client generation should be independent from our frontend technology.

***In GitLab by @xit-paul-reinhardt on Sep 6, 2024, 08:42 GMT+2:***

Basically the idea was to use the same node version everywhere and that should be the current lts version. I agree that pinning in `renovate.json` is missing (maybe it's even possible to only allow the lts releases?).

***In GitLab by @thaase on Sep 6, 2024, 08:47 GMT+2:***

Renovate understands node LTS: https://docs.renovatebot.com/modules/versioning/node/

I agree that we should use node LTS in all places. We should also document this decision somewhere.

***In GitLab by @xit-mohamed on Sep 6, 2024, 11:39 GMT+2:***

Changes are collected in !258

**Comment by noheton** [comment]
**Discussion in GitLab:**

***In GitLab by @thaase on Sep 6, 2024, 08:35 GMT+2:***

Commented on [.gitlab/ci/jobs/build-nuxtend.yml line 1](https://github.com/noheton/shepard/compare/2cafb783d5f7984b3c6c2cbcf5bda03740c92d00...1422c5770ff3c29b5fd205187a9f13b10130b81c#diff-f48a7dc544cddef483d8e85b05d1b3c22185a3f4b0e82d5c2d6a869a4fe247caR1)

https://github.com/noheton/shepard/blob/1422c5770ff3c29b5fd205187a9f13b10130b81c/.gitlab/ci/jobs/build-nuxtend.yml#L1



```suggestion:-0+0
.prepare_npm:
```

I like `prepare_npm` better since this only installs the package.json which is not specific to nuxt.

***In GitLab by @xit-paul-reinhardt on Sep 6, 2024, 08:45 GMT+2:***

The reason for this rename was that in the `build-frontend.yml` and `build-nuxtend.yml` the `npm ci` command in `prepare_npm` is executed with a workspace flag. So the prepare scripts differ, but if they have the same name one of them overrides the other and leads to issues. 

We could either remove the workspace flag (that could make the jobs take more time) or add a comment to remove the `for_nuxt` once we removed the old frontend.

***In GitLab by @thaase on Sep 6, 2024, 08:49 GMT+2:***

Let's call it `prepare_nuxtend` and change it as soon as the old frontend is gone. This way, this is easy to find via search&replace

***In GitLab by @xit-mohamed on Sep 6, 2024, 11:39 GMT+2:***

Changes are collected in !258

**Comment by noheton** [comment]
**Discussion in GitLab:**

***In GitLab by @xit-paul-reinhardt on Sep 6, 2024, 08:46 GMT+2:***

Commented on [.gitlab/ci/jobs/build-frontend.yml line 2](https://github.com/noheton/shepard/compare/2cafb783d5f7984b3c6c2cbcf5bda03740c92d00...1422c5770ff3c29b5fd205187a9f13b10130b81c#diff-3718881fd59c6779119d444a9e70255fe58a1be04148627b8f5055db9bac671eR2)

https://github.com/noheton/shepard/blob/1422c5770ff3c29b5fd205187a9f13b10130b81c/.gitlab/ci/jobs/build-frontend.yml#L2



Would a slim image also work here (as well as in `build-nuxtend.yml`?

***In GitLab by @xit-mohamed on Sep 6, 2024, 11:40 GMT+2:***

yes, changes are collected in !258

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 6, 2024, 11:40 GMT+2:***

resolved all threads

**Comment by noheton** [comment]
**Discussion in GitLab:**

***In GitLab by @xit-paul-reinhardt on Sep 9, 2024, 11:19 GMT+2:***

Commented on [.gitlab/ci/mr.gitlab-ci.yml line 15](https://github.com/noheton/shepard/compare/2cafb783d5f7984b3c6c2cbcf5bda03740c92d00...1422c5770ff3c29b5fd205187a9f13b10130b81c#diff-976891847b9b0426bd8b6e775aa8a83b35778a6b5939531e7b095284c55137a7R15)

https://github.com/noheton/shepard/blob/1422c5770ff3c29b5fd205187a9f13b10130b81c/.gitlab/ci/mr.gitlab-ci.yml#L15



I think this should also be included in the develop pipeline

***In GitLab by @xit-mohamed on Sep 9, 2024, 16:21 GMT+2:***

Done in !262

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Sep 9, 2024, 16:21 GMT+2:***

resolved all threads
